src/Razor/Microsoft.AspNetCore.Razor.Language/src/Extensions/DesignTimeDirectiveTargetExtension.cs

```diff
@@ -192,6 +192,35 @@ private void WriteDesignTimeDirectiveToken(CodeRenderingContext context, Directi
                             context.CodeWriter.WriteLine(";");
                         }
                         break;
+                    case DirectiveTokenKind.GenericTypeConstraint:
+                        // We generate a generic local function with a generic parameter using the
+                        // same name and apply the constraints, like below.
+                        // The two warnings that we disable are:
+                        // * Hiding the class type parameter with the parameter on the method
+                        // * The function is defined but not used.
+                        // static void TypeConstraints_TParamName<TParamName>() where TParamName ...;
+                        context.CodeWriter.WriteLine("#pragma warning disable CS0693");
+                        context.CodeWriter.WriteLine("#pragma warning disable CS8321");
```

Comment:
Not sure if this would go against convention, but could we do:

```suggestion
                        context.CodeWriter.WriteLine("#pragma warning disable CS0693, CS8321");
```

Reply:
Yeah, we don't really have a convention, but we always generate 1 pragma per line, so I'm leaving it this way.